src/chainparams.h

```diff
@@ -34,6 +34,10 @@ struct CCheckpointData {
     double fTransactionsPerDay;
 };
 
+static const int DEFAULT_MAINNET_PORT = 8333;
+static const int DEFAULT_TESTNET_PORT = 18333;
+static const int DEFAULT_NOLNET_PORT = 9333;
```

Comment:
I disagree here - this is the correct approach for compile-time substitution of the fundamental types.  There will not even be memory allocated unless the address is taken, which it isn't of course.